nucypher/characters/base.py

```diff
@@ -305,19 +326,104 @@ def get_nodes_by_ids(self, node_ids):
     def write_node_metadata(self, node, serializer=bytes) -> str:
 
         try:
-            filename = "{}.node".format(node.checksum_public_address)  # TODO: Use common name
+            filename = "{}.node".format(node.checksum_public_address)
         except AttributeError:
             raise AttributeError("{} does not have a rest_interface attached".format(self))
 
         metadata_filepath = os.path.join(self.known_metadata_dir, filename)
-
         with open(metadata_filepath, "w") as f:
             f.write(serializer(node).hex())
+        self.log.info("Wrote new node metadata {}".format(metadata_filepath))
         return metadata_filepath
 
-    @abstractmethod
-    def learn_from_teacher_node(self, eager: bool = True):
-        raise NotImplementedError
+    def learn_from_teacher_node(self, eager=True):
+        """
+        Sends a request to node_url to find out about known nodes.
+        """
+        self._learning_round += 1
+
+        try:
+            current_teacher = self.current_teacher_node()
+        except self.NotEnoughTeachers as e:
+            self.log.warning("Can't learn right now: {}".format(e.args[0]))
+            return
+
+        rest_url = current_teacher.rest_interface  # TODO: Name this..?
+
+        # TODO: Do we really want to try to learn about all these nodes instantly?
+        # Hearing this traffic might give insight to an attacker.
+        if VerifiableNode in self.__class__.__bases__:
```

Comment:
Not that it matters, but the "official" interface is probably `isinstance(self, VerifiableNode)`

Reply:
Changed upstream in #470